src/python/pants/build_graph/build_graph.py

```diff
@@ -520,6 +520,13 @@ def resolve_address(self, address):
     :returns: The Addressable which address points to.
     """
 
+  @abstractmethod
+  def fork(self):
+    """Fork a buildgraph from scratch without copying its state.
```

Comment:
yea, `fork` or `clone` isn't accurate, this is to create an uninitialized graph of the same type.

let me think about it more.